biocentral_server/predict/triton_predictor.py

```diff
@@ -0,0 +1,329 @@
+"""Triton-based prediction adapter for protein property prediction.
+
+This module provides a Triton-backed implementation of the prediction interface,
+allowing predictions to be computed via Triton Inference Server instead of local ONNX.
+"""
+
+import asyncio
+from typing import Dict, List
+import torch
+import numpy as np
+
+from biotrainer.protocols import Protocol
+
+from ..utils import get_logger
+from ..server_management import (
+    TritonClientConfig,
+    TritonModelRouter,
+    create_triton_repository,
+)
+from .models.base_model import BaseModel, Prediction, ModelMetadata
+
+logger = get_logger(__name__)
+
+
+class TritonPredictor(BaseModel):
+    """Triton-based predictor that delegates inference to Triton Inference Server.
+
+    This class provides the same interface as BaseModel but uses Triton for inference
+    instead of local ONNX models. It maintains compatibility with existing prediction
+    workflows while leveraging Triton's optimizations.
+    """
+
+    def __init__(
+        self,
+        batch_size: int,
+        triton_model_name: str,
+        metadata: ModelMetadata,
+        label_maps: Dict[str, Dict[int, str]] = None,
+        delimiter: str = "",
+    ):
+        """Initialize Triton predictor.
+
+        Args:
+            batch_size: Batch size for predictions
+            triton_model_name: Name of the model in Triton repository
+            metadata: Model metadata
+            label_maps: Mapping from prediction_name to label mappings
+            delimiter: Delimiter for per-residue predictions
+        """
+        # Skip BaseModel __init__ since we don't load ONNX models
```

Comment:
I think then the base_model implementation of __init__ should be changed not to load the models (I can also do that later).